src/runtime/components/forms/FormGroup.vue

```diff
@@ -92,14 +95,17 @@ export default defineComponent({
     })
 
     const size = computed(() => ui.value.size[props.size ?? appConfig.ui.input.default.size])
+    const labelFor = computed(() => `${props.name}-${increment = increment < 1000000 ? increment + 1 : 0}`)
```

Comment:
```suggestion
    const labelFor = ref(`${props.name}-${increment = increment < 1000000 ? increment + 1 : 0}`)
```

This should be changed to a ref to allow write operations in `useFormGroup`

Reply:
Hi @romhml, I've changed how to update `labelFor` when the input already has an `id`.

from
```javascript
formGroup.labelFor.value = inputAttrs?.id ?? formGroup?.labelFor.value
```

to
```javascript
const labelFor = computed(() => inputAttrs?.id || formGroup?.labelFor.value)
```

It will ensure that our `labelFor` value is updated based on the new `id` attribute of `UInput` and the new `name` attribute of `UFormGroup`.

Reply:
I'm not able to test it right now but I think this won't update the `for=` attribute in the FormGroup if an id is provided

Reply:
@romhml, these are my testing results. I hope they can be helpful.

https://github.com/nuxt/ui/assets/138564696/f2321d41-a81d-477e-af38-cb37e8b55e07

Reply:
Here's an example of what I meant:

```vue
    <UFormGroup label="Email" name="email">
      <UInput id="foo" v-model="state.email" />
    </UFormGroup>
```

Results in the following:
```html
<label for="email-4" ...>Email</label>
<input id="foo" type="text" ...>
```

Instead of: 
```html
<label for="foo" ...>Email</label>
<input id="foo" type="text" ...>
```

Which breaks accessibility features and interactions (like clicking on the label to focus the input).

Reply:
Using
```ts
const labelFor = ref(`${props.name}-${increment = increment < 1000000 ? increment + 1 : 0}`)
```
allows to properly update the value in `useFormGroup` to handle explicit ids properly. Reactivity can then be achieved using `watch`, although I don't think it's needed here (can't see a lot of use cases)

Reply:
Sorry, @romhml, I missed the previous comment. I have updated it.

> ... although I don't think it's needed here (can't see a lot of use cases)

I assumed there was a bug when there is a field array, but that's not the case. I tested it with my new update, and the bug doesn't appear.